lib/Bio/EnsEMBL/Versioning/Pipeline/CheckCoordinateOverlap.pm

```diff
@@ -0,0 +1,125 @@
+use warnings;
+
+use parent qw/Bio::EnsEMBL::Versioning::Pipeline::Base/;
+
+use Bio::EnsEMBL::Utils::Exception qw/throw/;
+use Bio::EnsEMBL::Versioning::Broker;
+use Bio::EnsEMBL::Mongoose::Persistence::QueryParameters;
+use Bio::EnsEMBL::Mongoose::IndexReader;
+use Bio::EnsEMBL::Mongoose::Serializer::RDF;
+use Bio::EnsEMBL::Mongoose::Serializer::RDFCoordinateOverlap;
+use Bio::EnsEMBL::Mongoose::Taxonomizer;
+use Bio::EnsEMBL::Versioning::CoordinateMapper;
+use Bio::EnsEMBL::ApiVersion qw/software_version/;
+use File::Path qw/make_path/;
+use File::Spec;
+use IO::File;
+use Bio::EnsEMBL::Utils::IO qw/slurp_to_array/;
+
+
+sub fetch_input {
+  my $self = shift;
+  my $eg = $self->param('eg');
+  my $release;
+  if ($eg) {
+     $release = $self->param('eg_version');
+  } else {
+    $release = software_version;
+  }
+  my $species = $self->param('species');
+  
+  # Derive location of coordinate_overlap
+  my $indexer = Bio::EnsEMBL::Mongoose::IndexReader->new(species => $species);
+
+  $self->param('indexer',$indexer);
+  
+  # Define where the RDF output will go and create the folder
+  my $output_path = File::Spec->join( $self->param('base_path'), "xref_rdf_dumps", $self->param('run_id'), $species, 'coordinate_overlap');
+  if (!-d $output_path) {
+    make_path $output_path or die "Failed to create path: $output_path";
+  }
+  $self->param('output_path',$output_path);
+}
+
+sub run {
+  my ($self) = @_;
+
+  my $run_id = $self->param('run_id'); # Comes from hive param stack
+  my $core_dba = $self->get_DBAdaptor('core');
+  
+  my $indexer = $self->param('indexer');
+  my $output_path = $self->param('output_path');
+  my $species= $self->param('species');
+  
+  my $broker = Bio::EnsEMBL::Versioning::Broker->new;
+  my @sources = qw/refseq ucsc/;
+  
+  # Coordinate overlap is done only for refseq and ucsc
+  foreach my $source(@sources){
+    my $fh = IO::File->new($output_path.'/'.$source.'_coordinate_overlap.ttl','w') or throw("Failed to open ${source}_overlap.ttl for writing");
+    my $rdf_writer = Bio::EnsEMBL::Mongoose::Serializer::RDFCoordinateOverlap->new(handle => $fh, config_file => $ENV{MONGOOSE}.'/conf/manager.conf');
+
+    my $mapper = Bio::EnsEMBL::Versioning::CoordinateMapper->new;
+    
+    # for refseq we need to covert the rows from otherfeatures database to lucy index with coorinate info stored in the records. 
+    # for UCSC, we already have the index with coordinate info
+    if($source eq "refseq"){
+      my $other_dba = $self->get_DBAdaptor('otherfeatures');
+      next unless defined $other_dba;
+      my $temp_index_folder = $mapper->create_temp_index({'species' => $species, 'dba' => $other_dba, 'analysis_name' => $source."_import"});
+      if(defined $temp_index_folder && -e $temp_index_folder){
+        $mapper->calculate_overlap_score({'index_location' => $temp_index_folder , 'species' => $species, 'core_dba' => $core_dba, 'other_dba' => $other_dba,'rdf_writer' => $rdf_writer , 'source' => $source});
```

Comment:
Doesn't need fixing, but you don't need to quote left of the "fat comma" =>, as the operator quotes to the left for you. Only necessary when there are Perl sigils to protect.